### ISSUE TITLE
zap-api-java - ApiResponseElement casting error in constructor of ApiResponseSet (if child is also a set) 

### ISSUE THREAD

**Issue by snevesbarros** [zaproxy/zaproxy#3228]
I am unable to use client.httpSessions.sessions(website, "") method with the zap-api-java client as the xml response which the ZAP API returns contains a set within a set (e.g. for the tokens).

```xml
<sessions type="list">
   <session type="list">
      <name>Session 0</name>
      <tokens type="set">
         <token type="set">
            <name>JSESSIONID</name>
            <value>.....w02</value>
            <domain>.examplesite.com</domain>
            <path>/</path>
            <secure>false</secure>
         </token>
      </tokens>
      <messages_matched>1</messages_matched>
   </session>
</sessions>
```
The ApiResponseSet(Node node) constructor assumes the child is not a set (and tries to cast it to a ApiResponseElement) which then throws a casting exception.

**Comment by snevesbarros** [comment]
Having looked a bit further into the issue I believe the problem could be that tokens has type="set", but this should be changed to "list" (allowing the child set to be processed using ApiResponseFactory.getResponse(child) method in the ApiResponseList Constructor)

**Comment by thc202** [comment]
Well, it's a set because the tokens are unique. (In any case, changing the `ApiResponseSet` to handle any `ApiResponse` would avoid future problems.)

**Comment by thc202** [comment]
(Related to #2246.)

**Comment by snevesbarros** [comment]
@thc202 I don't understand "the tokens are unique" reference. Would changing the parent node "tokens" to type=list not fix this. E.g. as it is a list of tokens (and within this list each token will still have type=set)?

**Comment by snevesbarros** [comment]
@thc202 Ok get it now you can only have one token.

**Comment by thc202** [comment]
With unique I mean that there's only one token/cookie with the same name. We definitely can use a list for that, it's easier to consume but it will require a new ZAP release (and might break other clients implementations).

**Comment by snevesbarros** [comment]
Sweet! Thanks. 👍

**Comment by thc202** [comment]
Released in version 1.1.0 of Java ZAP API client (although a newer version, 1.1.1, is already available).
